mentat/code_file.py

```diff
@@ -53,6 +84,87 @@ def __init__(self, path: str | Path):
                 self.intervals = [Interval(0, math.inf)]
             else:
                 self.intervals = parse_intervals(split[1])
+                level = CodeMessageLevel.INTERVAL
+        self.level = level
+        self.diff = diff
+
+    def __repr__(self):
+        return (
+            f"CodeFile(fname={self.path.name}, intervals={self.intervals},"
+            f" level={self.level}, diff={self.diff})"
+        )
 
     def contains_line(self, line_number: int):
         return any([interval.contains(line_number) for interval in self.intervals])
+
+    async def _get_file_message(
+        self,
+        config: ConfigManager,
+        code_file_manager: "CodeFileManager",
+        parser: "Parser",
+    ) -> list[str]:
+        file_message: list[str] = []
+
+        # We always want to give GPT posix paths
+        abs_path = Path(config.git_root / self.path)
+        rel_path = Path(os.path.relpath(abs_path, config.git_root))
+        posix_rel_path = Path(rel_path).as_posix()
+        file_message.append(posix_rel_path)
+
+        if self.level in {CodeMessageLevel.CODE, CodeMessageLevel.INTERVAL}:
+            file_lines = code_file_manager.read_file(abs_path)
```

Comment:
I'd prefer the `read_file` function to be a function of the CodeFile object. We don't actually need any information from CodeFileManager and the CodeFile can also cache the lines.

Reply:
This looks odd now, but I'm anticipating that there will be a single instance of CodeFileManager coordinating read/write/cache/checksum across all the different Sessions, so that's why it's still included. Also, #123 makes this a lot cleaner.

Reply:
Makes sense. I wonder if eventually we'll want to keep an sqlite database and cache the checksum and embedding across runs of Mentat.

Reply:
> Makes sense. I wonder if eventually we'll want to keep an sqlite database and cache the checksum and embedding across runs of Mentat.

Yep that's what I'm thinking! Both embeddings as well as summaries or whatever expensive computation we do to better understand codebases.